caravel/storage/helpers.py

```diff
@@ -57,13 +58,18 @@ def run_query(query=""):
     words = words[:5] # TODO: Raise once we know the approximate cost.
 
     # Retrieve the keys for entities that match all terms.
-    shards = [set(fetch_shard(entities.fold_query_term(w))) for w in words]
+    shards = [fetch_shard(entities.fold_query_term(w)) for w in words]
```

Comment:
We need to keep the keys in order of posting date; set uses hashing, which removes order.